scripts/machine_translation/inference_transformer.py

```diff
@@ -0,0 +1,286 @@
+                         '(using single gpu is suggested)')
+parser.add_argument('--model_parameter', type=str, default=' ', required=True,
+                    help='model parameter for inference, must be provided.')
+
+args = parser.parse_args()
+logging_config(args.save_dir)
+logging.info(args)
+
+# data process
+data_train, data_val, data_test, val_tgt_sentences, test_tgt_sentences, src_vocab, tgt_vocab \
+    = dataprocessor.load_translation_data(dataset=args.dataset, bleu=args.bleu, args=args)
+
+dataprocessor.write_sentences(test_tgt_sentences, os.path.join(args.save_dir, 'test_gt.txt'))
+
+data_train = data_train.transform(lambda src, tgt: (src, tgt, len(src), len(tgt)), lazy=False)
+data_val = gluon.data.SimpleDataset([(ele[0], ele[1], len(ele[0]), len(ele[1]), i)
+                                     for i, ele in enumerate(data_val)])
+data_test = gluon.data.SimpleDataset([(ele[0], ele[1], len(ele[0]), len(ele[1]), i)
+                                      for i, ele in enumerate(data_test)])
+
+data_train_lengths, data_val_lengths, data_test_lengths = [dataprocessor.get_data_lengths(x)
+                                                           for x in
+                                                           [data_train, data_val, data_test]]
+
+detokenizer = nlp.data.SacreMosesDetokenizer()
+
+# model prepare
+ctx = [mx.cpu()] if args.gpus is None or args.gpus == '' else \
+    [mx.gpu(int(x)) for x in args.gpus.split(',')]
+num_ctxs = len(ctx)
+
+if args.src_max_len <= 0 or args.tgt_max_len <= 0:
+    max_len = np.max(
+        [np.max(data_train_lengths, axis=0), np.max(data_val_lengths, axis=0),
+         np.max(data_test_lengths, axis=0)],
+        axis=0)
+
+if args.src_max_len > 0:
+    src_max_len = args.src_max_len
+else:
+    src_max_len = max_len[0]
+if args.tgt_max_len > 0:
+    tgt_max_len = args.tgt_max_len
+else:
+    tgt_max_len = max_len[1]
+
+encoder, decoder = get_transformer_encoder_decoder(units=args.num_units,
+                                                   hidden_size=args.hidden_size,
+                                                   dropout=args.dropout,
+                                                   num_layers=args.num_layers,
+                                                   num_heads=args.num_heads,
+                                                   max_src_length=max(src_max_len, 500),
+                                                   max_tgt_length=max(tgt_max_len, 500),
+                                                   scaled=args.scaled)
+model = NMTModel(src_vocab=src_vocab, tgt_vocab=tgt_vocab, encoder=encoder, decoder=decoder,
+                 share_embed=args.dataset != 'TOY', embed_size=args.num_units,
+                 tie_weights=args.dataset != 'TOY', embed_initializer=None, prefix='transformer_')
+
+param_name = args.model_parameter
+if (not os.path.exists(param_name)):
+  download("https://drive.google.com/open?id=1588i6OoaL8qC0K8gI3p2iFOYY5AEuRIN", fname=param_name)
```

Comment:
Can you add a warning that the provided file does not exist, and the download will happen?

Reply:
@eric-haibin-lin I've created a new commit to address your comments :).
And I think it's better to host this params on s3 to keep align with other dataset/param files.